File: branches/pre-scala-migration/src/java/org/codehaus/testdox/intellij/TestElement.java
package org.codehaus.testdox.intellij;

import javax.swing.Icon;

import com.intellij.openapi.actionSystem.Presentation;
import com.intellij.psi.PsiElement;

public interface TestElement extends Comparable {

    PsiElement getPsiElement();

    boolean jumpToPsiElement();

    String getDisplayString();

    Icon getIcon();

    void updatePresentation(Presentation presentation);

    void rename(TestDoxController testDoxController);

    void delete(TestDoxController testDoxController);
}

File: src/scala/org/codehaus/testdox/intellij/TestElement.scala
package org.codehaus.testdox.intellij

import javax.swing.Icon;

import com.intellij.openapi.actionSystem.Presentation
import com.intellij.psi.PsiElement

trait TestElement extends Ordered[TestElement] {

  def psiElement: PsiElement

  def jumpToPsiElement(): Boolean

  def displayString: String

  def icon: Icon

  def update(presentation: Presentation)

  def rename(controller: TestDoxController)

  def delete(controller: TestDoxController)

  def compare(that: TestElement): Int
}
